src/esp/gfx/PbrPrecomputedMapShader.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "PbrPrecomputedMapShader.h"
+#include "PbrTextureUnit.h"
+
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Containers/Reference.h>
+#include <Corrade/Utility/Debug.h>
+#include <Corrade/Utility/DebugStl.h>
+#include <Corrade/Utility/FormatStl.h>
+#include <Corrade/Utility/Resource.h>
+#include <Magnum/GL/Context.h>
+#include <Magnum/GL/Extensions.h>
+#include <Magnum/GL/Shader.h>
+#include <Magnum/GL/Version.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Matrix4.h>
+
+#include <initializer_list>
+#include <sstream>
+
+// This is to import the "resources" at runtime. When the resource is
+// compiled into static library, it must be explicitly initialized via this
+// macro, and should be called *outside* of any namespace.
+static void importShaderResources() {
+  CORRADE_RESOURCE_INITIALIZE(ShaderResources)
+}
+
+namespace Mn = Magnum;
+namespace Cr = Corrade;
+
+namespace esp {
+namespace gfx {
+
+PbrPrecomputedMapShader::PbrPrecomputedMapShader(Flags flags) : flags_(flags) {
+  CORRADE_ASSERT(
+      !((flags_ & Flag::IrradianceMap) && (flags_ & Flag::PrefilteredMap)),
+      "PbrPrecomputedMapShader::PbrPrecomputedMapShader: "
+      "Flag:S:IrradianceMap and "
```

Comment:
```suggestion
      "Flag::IrradianceMap and "
```

(i get this *very* often myself as well, <kbd>shift</kbd> <kbd>s</kbd> pressed instead of <kbd>ctrl</kbd> :laughing:)